modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -432,6 +453,10 @@ public abstract class Base${schemaName}ResourceImpl
 			throw new UnsupportedOperationException("This method needs to be implemented");
 		}
 
+		protected Long getPermissionCheckerResourceId(Object id) throws Exception {
+			return GetterUtil.getLong(id);
```

Comment:
Is there any check before to see if schemaVarNameId exists and it is something that can be converted to an Id? I know that is the liferay convention but I'm thinking in an external use case.